src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs

```diff
@@ -385,6 +404,9 @@ public override void OnDestroy()
 			{
 				base.OnDestroy();
 				FireAnimationEnded();
+
+				// SAFETY: If destroyed before OnStart completed, fire PresentationCompleted to prevent deadlock
+				FirePresentationCompleted();
```

Comment:
I'm not sure if it's a good idea to complete the TCS, I would say the best here is to Cancel it